tests/integration/test_agent.py

```diff
@@ -15,6 +16,17 @@
 
 workspace_base = os.getenv('WORKSPACE_BASE')
 
+# make sure we're testing in the same folder of an existing config.toml
+if os.path.exists('config.toml'):
+    config = AppConfig()
+    load_from_toml(config, 'config.toml')
+    if config and config.workspace_base and config.workspace_base != workspace_base:
+        if os.path.exists(config.workspace_base) and os.access(
+            config.workspace_base, os.W_OK
+        ):
+            print(f'Setting workspace_base to {config.workspace_base}')
+            workspace_base = config.workspace_base
+
```

Comment:
@li-boxuan this is one thing that took me hours to find and resolve:
The runtime loads values from the config.toml file, which may have a different workspace_base than what regenerate sets. This resulted in files being read and written in 2 different folders, thus multiple tests failing, like the "bad.txt" one.

Reply:
That's very weird, because the script sets environment variable `WORKSPACE_BASE`, which would override the value in `config.toml`.

Reply:
I just tried and it worked as expected. I set `workspace_base="./workspace_base_in_toml"` in my `config.toml` and ran the script. I printed out `workspace_base` after L17 and the value indeed was `_test_workspace` (defined in the script).

Reply:
I am using Mac btw. Not sure if we have some os-dependent behavior somewhere

Reply:
> I am using Mac btw. Not sure if we have some os-dependent behavior somewhere

That is what I fear with WSL right now.
Could you try this on your end if the file-related tests work with the config.toml file pointing to some different folder, please?
`TEST_ONLY=true ONLY_TEST_AGENT="CodeActAgent" ./tests/integration/regenerate.sh`
In my case, I have in config.toml:
`workspace_base="/mnt/d/github/workspace"`
which is one folder "below" my OpenDevin repo.

Reply:
I tried, and it worked. I set `workspace_base="/mnt/d/github/workspace"` in my `config.toml` too, which is a non-existent path.

What do you see when you run `echo $WORKSPACE_BASE $WORKSPACE_MOUNT_PATH`? I am guessing it's non-empty, otherwise I cannot explain it.

Reply:
I did an echo in regenerate whilst testing and both were set to
`/mnt/d/github/OpenDevin/_test_workspace`
before Run().
Do you think I should remove that block in regenerate.sh again or see side-effects with it?

Reply:
Nope doesn’t sound necessary. @enyst is there anything you can think of that might affect the order of loading configs?

Reply:
There is a thing that happens: config is loaded twice. The second time is when browser_env starts... I have not caught actual errors because of it, but it does mean there's a moment in time when things have started, including sandbox, but config is inconsistent until it's done reloading. Also raises the question what env does that process get.

A shot in the dark here... how about `persist_sandbox` var, do you have it set to true @tobitege ?

Reply:
Now that you mention it, I do have that set to true in my config.toml.
I guess I should try with false then? 😁

Reply:
I don't think it's tested with integration tests? I would set it to false, and clean up both containers and (opendevin) images from docker to be sure.

Reply:
So yeah that might very likely be the curse.